cpp/src/IceSSL/SChannelTransceiverI.cpp

```diff
@@ -753,7 +847,10 @@ SChannel::TransceiverI::initialize(IceInternal::Buffer& readBuffer, IceInternal:
     }
     catch(const Ice::SecurityException&)
     {
+        _trustError = IceSSL::ICE_ENUM(TrustError, HostNameMismatch);
         _verified = false;
+        ICE_DYNAMIC_CAST(ExtendedConnectionInfo, info)->errorCode = IceSSL::ICE_ENUM(TrustError, HostNameMismatch);
+        info->verified = false;
```

Comment:
Why do we set these since the info is no longer used after throwing?

Reply:
We don't throw if `getVerifyPeer() == 0` and pass this info object to `verifyPeer` call below